tests/secrets/mkdir.test.ts

```diff
@@ -94,6 +95,106 @@ describe('commandMkdir', () => {
       expect(stat2.isDirectory()).toBeTruthy();
     });
   });
+  test('should fail without recursive set', async () => {
+    const vaultName = 'vault' as VaultName;
+    const dirName1 = 'dir1';
+    const dirName2 = 'dir2';
+    const dirNameNested = `${dirName1}/${dirName2}`;
+    const vaultId = await polykeyAgent.vaultManager.createVault(vaultName);
+
+    command = [
+      'secrets',
+      'mkdir',
+      '-np',
+      dataDir,
+      `${vaultName}:${dirNameNested}`,
+    ];
+
+    const result = await testUtils.pkStdio([...command], {
+      env: { PK_PASSWORD: password },
+      cwd: dataDir,
+    });
+    expect(result.exitCode).toBe(0);
+    expect(result.stderr).toBeDefined();
```

Comment:
An exit code of `0` indicates that the command succeeded. Any non-zero code is a command failure. Also keep in mind any messaging, error or otherwise comes out on `stderr`. The existence of output on `stderr` isn't an indication of an error.

Reply:
> The existence of output on `stderr` isn't an indication of an error.

This wasn't to check for errors, but to make sure the warning is printed. I guess I can keep the `stderr` check in only one of the tests and use explicit `toInclude` or similar check if the error string exists in `stderr`.

> An exit code of `0` indicates that the command succeeded. Any non-zero code is a command failure.

I haven't implemented this yet. I will implement it so that the program will exit with a non-zero exit code if an error occurs by exiting the process using an exit command.

Reply:
I've been trying to solve this issue. Setting `process.exitCode` does not work and the process always returns an exit code of zero. We may need to have a chat about this.

Reply:
While not fully compliant, we will instead be throwing an error at the end if a directory operation failed. That is not the case with standard mkdir, but this is a side-effect of Polykey CLI's behaviour.

The command will now look like this:

```console
[aryanj@matrix-34xx:~]$ pk secrets mkdir vault:test vault:fail/test vault:test2
WARNING: Empty directories will not be affected by modifications to the vault state (version change, sharing/cloning, etc.)
NOENT: cannot create directory test: No such secret or directory
ErrorPolykeyCLIMakeDirectory: Failed to create one or more directories

[aryanj@matrix-34xx:~]$ echo $?
1
```